lib/cli-engine/formatters/text-table.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * @fileoverview Optimized version of the `text-table` npm module to improve performance by replacing inefficient regex-based
+ * whitespace trimming with a modern built-in method.
+ *
+ * @author Nitin Kumar
+ *
+ * This modification addresses a performance issue reported in https://github.com/eslint/eslint/issues/18709
+ *
+ * The `text-table` module is published under the MIT License. For the original source, refer to:
+ * https://www.npmjs.com/package/text-table.
+ */
+
+"use strict";
+
+function dotindex(c) {
+    const m = /\.[^.]*$/u.exec(c);
+
+    return m ? m.index + 1 : c.length;
+}
+
+function reduce(xs, f, init) {
+    if (xs.reduce) {
+        return xs.reduce(f, init);
+    }
+    let i = 0;
+    const acc = arguments.length >= 3 ? init : xs[i++];
+
+    for (; i < xs.length; i++) {
+        f(acc, xs[i], i);
+    }
+    return acc;
+}
+
+function forEach(xs, f) {
+    if (xs.forEach) {
+        return xs.forEach(f);
+    }
+    for (let i = 0; i < xs.length; i++) {
+        f.call(xs, xs[i], i);
+    }
+}
+
+function map(xs, f) {
+    if (xs.map) {
+        return xs.map(f);
+    }
+    const res = [];
+
+    for (let i = 0; i < xs.length; i++) {
+        res.push(f.call(xs, xs[i], i));
+    }
+    return res;
+}
+
+module.exports = function(rows_, opts) {
+    if (!opts) {
+        opts = {};
+    }
+    const hsep = typeof opts.hsep === "undefined" ? "  " : opts.hsep;
```

Comment:
The `stylish` formatter only ever calls to `table()` once. I think we can remove dead code in here by hardcoding in only what's actually used. For example, `opts` is always defined, and `hsep` can just be `"  "`.

Reply:
Updated 👍🏻